src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -114,76 +120,91 @@ private void OnProjectsChanged(ConnectionInformation connection, ProjectInformat
 
         #region Workflow steps
 
-        internal /* for testing purposes */ void ConnectionStep(IProgressController controller, CancellationToken cancellationToken, ConnectionInformation connection,
-            IProgressStepExecutionEvents notifications)
+        internal /* for testing purposes */ async Task ConnectionStepAsync(ConnectionInformation connection,
+            IProgressController controller, IProgressStepExecutionEvents notifications, CancellationToken cancellationToken)
         {
             this.host.ActiveSection?.UserNotifications?.HideNotification(NotificationIds.FailedToConnectId);
-            this.host.ActiveSection?.UserNotifications?.HideNotification(NotificationIds.BadServerPluginId);
+            this.host.ActiveSection?.UserNotifications?.HideNotification(NotificationIds.BadSonarQubePluginId);
 
             notifications.ProgressChanged(connection.ServerUri.ToString());
 
-            notifications.ProgressChanged(Strings.ConnectionStepValidatinCredentials);
-            if (!this.host.SonarQubeService.AreCredentialsValid(connection, cancellationToken))
+            try
             {
-                AbortWithMessage(notifications, controller, cancellationToken);
-                return;
-            }
+                notifications.ProgressChanged(Strings.ConnectionStepValidatinCredentials);
+                await this.host.SonarQubeService.ConnectAsync(connection, cancellationToken); // TODO: Handle errors
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)